service/shadowsocks.go

```diff
@@ -39,14 +36,15 @@ type ShadowsocksConnMetrics interface {
 }
 
 type ServiceMetrics interface {
-	UDPMetrics
+	AddOpenUDPAssociation(conn net.Conn) UDPAssociationMetrics
 	AddOpenTCPConnection(conn net.Conn) TCPConnMetrics
 	AddCipherSearch(proto string, accessKeyFound bool, timeToCipher time.Duration)
 }
 
 type Service interface {
```

Comment:
This interface is confused, since it's conflates mutually exclusive behavior.
We shouldn't have to implement the behavior for caddy in the outline-ss-server and vice-versa.

I don't think this interface is contributing much. Can we get rid of it and have separate interfaces instead?
Or define a Service interface in each specific apps instead?

`NewConnAssociation(conn)` can be replaced with a `HandleAssociation(conn)`